tests/acceptance/pageObjects/FilesPageElement/filesList.js

```diff
@@ -477,25 +477,31 @@ module.exports = {
      * @returns {Array} array of sharing indicator
      */
     getShareIndicatorsForResource: async function(fileName, sharingIndicatorExpectedToBeVisible) {
-      const resourceRowXpath = this.getFileRowSelectorByFileName(fileName)
-      const shareIndicatorsXpath =
-        resourceRowXpath + this.elements.shareIndicatorsInFileRow.selector
+      let resourceRowXpath = this.getFileRowSelectorByFileName(fileName)
+      let shareIndicatorsXpath = resourceRowXpath + this.elements.shareIndicatorsInFileRow.selector
       const indicators = []
       await this.waitForFileVisible(fileName)
       const {
         waitForNegativeConditionTimeout,
         waitForConditionTimeout,
         waitForConditionPollInterval
       } = client.globals
-      await this.waitForElementVisible({
-        selector: shareIndicatorsXpath,
-        locateStrategy: this.elements.shareIndicatorsInFileRow.locateStrategy,
-        abortOnFailure: false,
-        timeout: sharingIndicatorExpectedToBeVisible
-          ? waitForConditionTimeout
-          : waitForNegativeConditionTimeout,
-        pollInterval: waitForConditionPollInterval
-      })
+      try {
+        await this.waitForElementVisible({
+          selector: shareIndicatorsXpath,
+          locateStrategy: this.elements.shareIndicatorsInFileRow.locateStrategy,
+          abortOnFailure: false,
+          timeout: sharingIndicatorExpectedToBeVisible
+            ? waitForConditionTimeout
+            : waitForNegativeConditionTimeout,
+          pollInterval: waitForConditionPollInterval
+        })
+      } catch (error) {
+        console.log('\n Retrying with new reference \n')
+        resourceRowXpath = this.getFileRowSelectorByFileName(fileName)
+        shareIndicatorsXpath = resourceRowXpath + this.elements.shareIndicatorsInFileRow.selector
+        await this.waitForFileVisible(fileName)
+      }
```

Comment:
I am not sure we should do that. And also the error is thrown by
`508: this.api.elementIdAttribute(element.ELEMENT, 'data-test-indicator-type', attr => {`

what I suspect is, the DOM is refreshed due to the preview of the file causing the element reference to change.
so I suggest is to use `@disablePreviews` tag in `webUISharingInternalUsersSharingIndicator/shareWithUsers.feature` file